mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -130,6 +130,25 @@ class GaussianMixtureModel private[ml] (
   }
 
   /**
+   * Return the total log-likelihood for this model on the given data.
+   */
+  @Since("2.2.0")
+  def computeLogLikelihood(dataset: Dataset[_]): Double = {
```

Comment:
I have a question here, should we provide the final ```logLikelihood``` of the model in its summary as well? Since lots of users will use it to evaluate the current model, that they don't need to take another pass on data.

This will expose a public API, cc @jkbradley @sethah @srowen @MLnick to discuss the API.